thunder/core/transform_common.py

```diff
@@ -750,11 +752,40 @@ def computation(x):
         computation_trace: A computation trace created by ``interpreter``.
         orig_to_view_swap_map:
     """
+    from thunder.core.utils import consumers
     from thunder.torch import _inplace_to_out_of_place
 
-    if not any(is_functionalizable(bsym) for bsym in computation_trace.bound_symbols):
+    consumer_map = consumers(computation_trace)
+    bsym_to_idx: dict[BoundSymbol, int] = {}
+    idx_to_bsym: dict[int, BoundSymbol] = {}
+    any_functionalizable: bool = False
+    for idx, bsym in enumerate(computation_trace.bound_symbols):
+        if is_functionalizable(bsym):
+            any_functionalizable = True
+        bsym_to_idx[bsym] = idx
+        idx_to_bsym[idx] = bsym
+
+    if not any_functionalizable:
         return []
 
+    # Step 0: Detect and error out aliased func args/kwargs. sparse tensors are ignored.
+    flat_args, _ = tree_flatten((computation_trace.args, computation_trace.kwargs))
```

Comment:
I don't think we should need the tree flatten here (at least for the general jit)?

Reply:
It's still allowed in Thunder to construct a trace with non-empty dictionary for keyword arguments. All transforms should be generic when possible and error out when it's designed to work with just one specially constructed trace.